connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -387,4 +402,50 @@ boolean isHeartbeatTopic(String topic) {
     String formatRemoteTopic(String topic) {
         return replicationPolicy.formatRemoteTopic(sourceAndTarget.source(), topic);
     }
+
+    // Writing the translated offsets to the __consumer_offsets topic in the destination cluster,
+    // as long as no consumers in that group are connected to the destination cluster
+    private void syncCommitOffset() throws InterruptedException, ExecutionException {
+        MirrorClient mirrorClient = new MirrorClient(config.targetConsumerConfig());
+        Map<String, Map<TopicPartition, OffsetAndMetadata>> allOffsets =
+            mirrorClient.remoteConsumerOffsets(config.sourceClusterAlias(), config.adminTimeout());
+
+        // track the consumer that are not actively consuming from the replicated topic with same consumer group id
+        Set<String> idleConsumers = new HashSet<>();
+        DescribeConsumerGroupsResult describedGroups = targetAdminClient.describeConsumerGroups(allOffsets.keySet());
+        for (Map.Entry<String, KafkaFuture<ConsumerGroupDescription>> group : describedGroups.describedGroups().entrySet()) {
+            ConsumerGroupDescription consumerGroupDescription = group.getValue().get();
+            if (consumerGroupDescription.state().equals(ConsumerGroupState.EMPTY)) {
+                idleConsumers.add(consumerGroupDescription.groupId());
+            }
+
+        }
+
+
+        Map<String, Object> consumerConfig = config.targetConsumerConfig();
+        log.info("Syncing consumer offsets for non-active consumers in target: {}", idleConsumers);
+        for (String consumerGroupId : idleConsumers) {
+            boolean sync = true;
+            // do not sync up consumer offset if the translated offset is smaller than the latest
+            // consumer offset in the target
+            for (Entry<TopicPartition, OffsetAndMetadata> entry : targetAdminClient.listConsumerGroupOffsets(consumerGroupId)
+                                                                                   .partitionsToOffsetAndMetadata().get().entrySet()) {
+                long currentOffsetInTarget = entry.getValue().offset();
+                long translatedOffsetInSource = allOffsets.get(consumerGroupId).get(entry.getKey()).offset();
+                if (currentOffsetInTarget > translatedOffsetInSource) {
+                    sync = false;
+                    break;
+                }
+            }
+            if (sync) {
+                consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, consumerGroupId);
+                KafkaConsumer<byte[], byte[]> consumerOffsetSync = MirrorUtils
+                                                                       .newConsumer(consumerConfig);
+                consumerOffsetSync
+                    .assign(new ArrayList<>(allOffsets.get(consumerGroupId).keySet()));
+                consumerOffsetSync.commitSync(allOffsets.get(consumerGroupId));
+                consumerOffsetSync.close();
```

Comment:
We can use the new reset offsets API for this: https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=97551484

Reply:
new reset offsets API looks awesome, will make this change in the next iteration